src/main/scala/beam/physsim/jdeqsim/JDEQSimRunner.scala

```diff
@@ -78,14 +78,15 @@ class JDEQSimRunner(
 
     val maybeEventWriter = if (writeEvents) {
       val writer = PhysSimEventWriter(beamServices, jdeqsimEvents)
+      beamServices.matsimServices.addControlerListener(writer)
```

Comment:
What is the reason to add writer to the controller listener? It feels that at this point it will wait to finish all the events and close the writer: https://github.com/LBNL-UCB-STI/beam/blob/6fd11c820b34da1f9667a8480308c28326495257/src/main/scala/beam/physsim/jdeqsim/JDEQSimRunner.scala#L105-L108

Reply:
It's required to prevent exiting BEAM before everything is written to disk.

Reply:
@dimaopen now I got it, it is because you made `PhysSimEventWriter` asynchronous?